logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -151,13 +151,14 @@ def load_available_plugins
       require "logstash/plugins/builtin"
 
       GemRegistry.logstash_plugins.each do |plugin_context|
-        if plugin_context.spec.metadata.key?('java_plugin')
-          jar_files = plugin_context.spec.files.select {|f| f =~ /.*\.jar/}
+        if plugin_context.spec.metadata.key?('java_plugin') && LogStash::SETTINGS.get_value('pipeline.plugin_classloaders')
```

Comment:
While this bit of code effectively populates an entry in `@java_plugins` for use when `pipeline.plugin_classloaders: true`, is there still value in still performing the validation when `pipeline.plugin_classloaders: false`? Or are we relying on a runtime exception when the plugin attempts to require a jar and it is not found on the require path?

My worry here is that effectively removing this validation from the default configuration will result in users building plugins that work _by accident_, but fail when Logstash is configured with plugin classloaders.

Reply:
That's fair. My intent was to allow the default (non-experimental) path to be as simple as possible, but I think you're right - removing the check opens the possibility of difficult to diagnose issues if a java plugin has been built and not packaged correctly.